src/components/ConfigEditor/AzureCredentialsForm.tsx

```diff
@@ -0,0 +1,218 @@
+      });
+    }
+
+    return opts;
+  }, [props.managedIdentityEnabled, props.oboEnabled]);
+
+  const onAuthTypeChange = (selected: SelectableValue<AzureAuthType>) => {
+    if (onCredentialsChange) {
+      const updated: AzureCredentials = {
+        ...credentials,
+        authType: selected.value || 'msi',
+      };
+      onCredentialsChange(updated);
+    }
+  };
+
+  const onAzureCloudChange = (selected: SelectableValue<string>) => {
+    if (
+      onCredentialsChange &&
+      (credentials.authType === 'clientsecret' || credentials.authType === 'clientsecret-obo')
+    ) {
+      const updated: AzureCredentials = {
+        ...credentials,
+        azureCloud: selected.value,
+      };
+      onCredentialsChange(updated);
+    }
+  };
+
+  const onTenantIdChange = (event: ChangeEvent<HTMLInputElement>) => {
+    if (
+      onCredentialsChange &&
+      (credentials.authType === 'clientsecret' || credentials.authType === 'clientsecret-obo')
+    ) {
+      const updated: AzureCredentials = {
+        ...credentials,
+        tenantId: event.target.value,
+      };
+      onCredentialsChange(updated);
+    }
+  };
+
+  const onClientIdChange = (event: ChangeEvent<HTMLInputElement>) => {
+    if (
+      onCredentialsChange &&
+      (credentials.authType === 'clientsecret' || credentials.authType === 'clientsecret-obo')
+    ) {
+      const updated: AzureCredentials = {
+        ...credentials,
+        clientId: event.target.value,
+      };
+      onCredentialsChange(updated);
+    }
+  };
+
+  const onClientSecretChange = (event: ChangeEvent<HTMLInputElement>) => {
+    if (
+      onCredentialsChange &&
+      (credentials.authType === 'clientsecret' || credentials.authType === 'clientsecret-obo')
+    ) {
+      const updated: AzureCredentials = {
+        ...credentials,
+        clientSecret: event.target.value,
+      };
+      onCredentialsChange(updated);
+    }
+  };
+
+  const onClientSecretReset = () => {
+    if (
+      onCredentialsChange &&
+      (credentials.authType === 'clientsecret' || credentials.authType === 'clientsecret-obo')
+    ) {
+      const updated: AzureCredentials = {
+        ...credentials,
+        clientSecret: '',
+      };
+      onCredentialsChange(updated);
+    }
+  };
+
+  return (
+    <div className="gf-form-group">
+      {authTypeOptions.length > 1 && (
+        <InlineField
+          label="Authentication"
+          labelWidth={18}
+          tooltip="Choose the type of authentication to Azure services"
+          htmlFor="azure-auth-type"
+        >
+          <Select
+            inputId="azure-auth-type"
+            className="width-15"
+            value={authTypeOptions.find((opt) => opt.value === credentials.authType)}
+            options={authTypeOptions}
+            onChange={onAuthTypeChange}
+          />
+        </InlineField>
+      )}
+      {credentials.authType === 'clientsecret-obo' && (
+        <>
+          {/* eslint-disable prettier/prettier */}
```

Comment:
why disabling prettier?

Reply:
Prettier is enforcing complete nonsense here

Reply:
It doesn't like the plain text here and tries to format it in a weird way